testsuite/tests/gc-roots/globroots.ml

```diff
@@ -45,23 +43,23 @@ module Test(G: GLOBREF) () = struct
     done
 
   let change () =
-    match Random.State.int (Domain.DLS.get random_state) 37 with
+    match Random.int 37 with
```

Comment:
Does this change preserve the behavior of the test? If I read things correctly, previous all domains would start from their own self-initialized seed, while the behavior of `Random.int 37` is to start from the static seed (independently in any domain).
I think that the change is "correct" in the sense that the new version is in-line with the behavior of the test on trunk. But thinking about this code made me realize that working with Random global state in multi-domain programs is not simple.

Reply:
In the case of this test, I think that the proposed change is the correct one:
- the new test coincides with the upstream version of the test
- the test is single-domain anyway, so the problematic correlations are not going to show up

Reply:
Although only executed in the `globroots` tests in a single-domain, I think the `Globroots` module is used in other tests with multiple domains. 
I see the `Globroots` module getting used by [globroots_parallel.ml](https://github.com/ocaml-multicore/ocaml-multicore/blob/b9afce3e9a474682f4003965cbc87bb0413e1e2d/testsuite/tests/gc-roots/globroots_parallel.ml) and [globroots_parallel_spawn_burn.ml](https://github.com/ocaml-multicore/ocaml-multicore/blob/b9afce3e9a474682f4003965cbc87bb0413e1e2d/testsuite/tests/gc-roots/globroots_parallel.ml).

Reply:
Ah, I had missed this, thanks!